masterData.ftl

```diff
@@ -86,6 +85,70 @@
         ]
       }
     },
+    "ResourceProfiles": {
+      "azure" : {
+        "microsoft.network" : { 
+          "applicationSecurityGroups" : {
+            "apiVersion" : "2019-04-01",
+            "type" : "Microsoft.Network/applicationSecurityGroups"
+          },
+          "routeTables" : {
+            "apiVersion" : "2019-02-01",
+            "type" : "Microsoft.Network/routeTables"
+          },
+          "routes" : {
+            "apiVersion" : "2019-02-01",
+            "type" : "Microsoft.Network/routeTables/routes"
+          },
+          "serviceEndpointPolicies" : {
+            "apiVersion" : "2019-02-01",
+            "type" : "Microsoft.Network/serviceEndpointPolicies"
+          },
+          "serviceEndpointPolicyDefinitions" : {
+            "apiVersion" : "2019-02-01",
+            "type" : "Microsoft.Network/serviceEndpointPolicies/serviceEndpointPolicyDefinitions"
+          },
+          "subnets" : {
+            "apiVersion" : "2019-02-01",
+            "type" : "Microsoft.Network/virtualNetworks/subnets"
+          },
+          "virtualNetworks" : {
+            "apiVersion" : "2019-02-01",
+            "type" : "Microsoft.Network/virtualNetworks"
+          },
+          "virtualNetworkPeerings" : {
+            "apiVersion" : "2019-02-01",
+            "type" : "Microsoft.Network/virtualNetworks/virtualNetworkPeerings"
+          },
+          "networkSecurityGroups" : {
+            "apiVersion" : "2019-02-01",
+            "type" : "Microsoft.Network/networkSecurityGroups"
+          },
+          "securityRules" : {
+            "apiVersion" : "2019-04-01",
+            "type" : "Microsoft.Network/networkSecurityGroups/securityRules"
+          },
+          "networkWatchers" : {
+            "apiVersion" : "2019-04-01",
+            "type" : "Microsoft.Network/networkWatchers"
+          }
+        },
+        "microsoft.storage" : {
+          "account" : {
+            "apiVersion" : "2019-04-01",
+            "type" : "Microsoft.Storage/storageAccounts"
+          },
+          "blob" : {
+            "apiVersion" : "2019-04-01",
+            "type" : "Microsoft.Storage/storageAccounts/blobServices"
+          },
+          "container" : {
+            "apiVersion" : "2019-04-01",
+            "type" : "Microsoft.Storage/storageAccounts/blobServices/containers"
+          }
+        }
+      }
+    }
```

Comment:
I like the idea but not sure if a masterdata profile is the best spot for them. Reference data ( which covers profiles in https://github.com/codeontap/gen3/pull/999 ) is available to users to modify. Given that the version is locked in the resource templates I think this might not be appropriate for that 

Maybe a global variable set in 
https://github.com/codeontap/gen3-azure/blob/2e0e7a75a4478111f94d7db4667c87843fd27e41/services/resource.ftl 

Which contains the ResourceProfiles section along with a macro that adds to it which is invoked in the resource definition files would work? This would keep the definition of the version and resources together with their actual implementation. This is a similar approach to how we do the output mappings at the moment, but would be vender specific

Reply:
@roleyfoley I have removed from MasterData and implimented a global variable ``azureResourceProfiles`` that will be added to for every Service loaded.

I have then updated the armResource macro to use these new profiles for the resource type/api values. If you're happy with this approach I will update the same to the Network component's PR (#32)